libgadget/blackhole.c

```diff
@@ -524,8 +524,9 @@ blackhole(const ActiveParticles * act, double atime, Cosmology * CP, ForceTree *
     priv->hubble = hubble_function(CP, atime);
     priv->GravInternal = CP->GravInternal;
 
-    /* Build the queue once, since it is really 'all black holes' and similar for all treewalks*/
-    treewalk_build_queue(tw_dynfric, act->ActiveParticle, act->NumActiveParticle, 0);
+    /* Build the queue once, since it is really 'all black holes' and similar for all treewalks.
+     * Allocate high so we can keep the tree we build after making the list.*/
+    treewalk_build_queue(tw_dynfric, act->ActiveParticle, act->NumActiveParticle, 0, 1);
```

Comment:
Question 1: Does the force_tree_rebuild_mask alloc-high, such that build_queue must also alloc-high?

Question 2: The alloc_high knob adds a fragile coupling between build_queue and caller. 
This seems to be the only user of the knob. What if  we do an alloc-high here (on the caller side), and memcpy the queue?

Reply:
1) The force_tree_rebuild_mask is alloc-low but I want to let the built force tree persist past the end of blackhole so I can reuse it for the winds. So I then can't free build_queue in time.
2) It isn't a bad idea. I was just trying to avoid a memcpy, but this is going to be a small one.

Reply:
I usually weight a small memcpy cheaper than a new function argument. :P

Reply:
It is fair :)